src/server/src/routes/student/index.ts

```diff
@@ -142,6 +143,68 @@ router.get(
     }
 );
 
+/******************************************************************************
+ *            GET Request - Search - /api/student/search"
+ ******************************************************************************/
+
+interface StudentSearchRequest extends Request {
+    query: {
+        firstName?: string;
+        lastName?: string;
+        email?: string;
+        sid?: string;
+        departmentIds?: string[];
+        classStandings?: classStandings[];
+        page: string;
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/search',
+    passport.authenticate('jwt', { session: false }),
+    validationMiddleware({ querySchema: studentSearchSchema }),
+    async (req: StudentSearchRequest, res: Response) => {
+        const { departmentIds, page, numOfItems } = req.query;
+        let { firstName, lastName, email, sid, classStandings } = req.query;
+
+        if (!firstName) firstName = "";
+        if (!lastName) lastName = "";
+        if (!email) email = "";
+        if (!sid) sid = "";
+        var departmentIdInts = [];
+        // Pass -1 when the input is empty because it causes a sql parse error
+        // when we pass in an empty array.
+        if (!departmentIds) departmentIdInts = [-1];
+        else {
+            for (const id of departmentIds) {
+                departmentIdInts.push(parseInt(id));
+            }
+        }
+        if (!classStandings) classStandings = ['Freshman', 'Sophomore', 'Junior', 'Senior'];
```

Comment:
I noticed this array used here maybe it should be declared in entities/students. In that file, please declare and export an array classStandingValues that stores 'Freshman', 'Sophomore', 'Junior', and 'Senior'. Then, import it in this file and use it in your if branch. Also, please update the entity Student class to use that new array (line 36 on entities/student.tsx).

Reply:
Also note that classStandings will never be undefined, so I think your if condition will have to be classStandings.length === 0

Reply:
How do I update the entity Student class to use the new array? Does this work?
classStandingsStrings = ['Freshman', 'Sophomore', 'Junior', 'Senior'];
classStandings = classStandingsStrings [0] | classStandingsStrings [1] | ...

Reply:
```
@Column({
        nullable: true,
        type: 'enum',
        enum: classStandingValues, // classStandingValues is the array that contains the class standings values
    })
    classStanding?: classStandings;
```

Reply:
Done.